influxql/emitter.go

```diff
@@ -66,6 +75,35 @@ func (e *Emitter) Emit() *models.Row {
 	}
 }
 
+// loadBuf reads in points into empty buffer slots.
+// Returns the next time/name/tags to emit for.
+func (e *Emitter) loadBuf() (t int64, name string, tags Tags) {
+	t = ZeroTime
+
+	for i := range e.itrs {
+		// Load buffer, if empty.
+		if e.buf[i] == nil {
+			e.buf[i] = e.readIterator(e.itrs[i])
+		}
+
+		// Skip if buffer is empty.
+		p := e.buf[i]
+		if p == nil {
+			continue
+		}
+
+		// Otherwise compare time/name/tags.
+		itrTime, itrName, itrTags := p.time(), p.name(), p.tags()
+		if t == ZeroTime ||
+			(e.ascending && ((itrTime < t) || (itrTime == t && itrName < name) || (itrTime == t && itrName == name && itrTags.ID() < tags.ID()))) ||
```

Comment:
I think these conditions are long enough that it might be better if they get moved into a small private function instead. That way the error condition reads more easily.